### ISSUE TITLE
Removed claim reappears at later block

### ISSUE THREAD

**Issue by edjroz** [pokt-network/pocket-core#1170]
**Describe the bug**
During testing of RC-0.5.2 we have spotted a possible data race in which a deleted claim reappeared at a later height which cause a chain halt because incorrect block proposal. This is only evident on a mixed setup environment with multiple releases coordinating 
- RC-0.5.1
- RC-0.5.0
- RC-0.5.2

**To Reproduce**
TBD

**Expected behavior**
A deleted claim should not be proposed on a future block.

**Additional context**
5.2 state before
```
                {
                    "evidence_type": "1",
                    "expiration_height": "207",
                    "from_address": "35df1afa3f2156cedaebe2e3a8e831b92c3a5ae3",
                    "header": {
                        "app_public_key": "60f18be9678468a6c5d2e02510f5de08185926e189ca00d57fd1291a15ffb538",
                        "chain": "0011",
                        "session_height": "3"
                    },
                    "merkle_root": {
                        "merkleHash": "c3Ak/+4lpSD5/Hw7ScN7O95zaJUfK6chTj6J4E/4wM4=",
                        "range": {
                            "lower": "0",
                            "upper": "18422676074773973113"
                        }
                    },
                    "total_proofs": "430"
                },
```

5.2 state after
```
                {
                    "evidence_type": "1",
                    "expiration_height": "207",
                    "from_address": "35df1afa3f2156cedaebe2e3a8e831b92c3a5ae3",
                    "header": {
                        "app_public_key": "60f18be9678468a6c5d2e02510f5de08185926e189ca00d57fd1291a15ffb538",
                        "chain": "0011",
                        "session_height": "3"
                    },
                    "merkle_root": {
                        "merkleHash": "c3Ak/+4lpSD5/Hw7ScN7O95zaJUfK6chTj6J4E/4wM4=",
                        "range": {
                            "lower": "0",
                            "upper": "18422676074773973113"
                        }
                    },
                    "total_proofs": "430"
                },
```

Log error from nodes, cannot agree on the state:
```
pocket-core-testnet2_1  | E[2020-10-21|15:07:26.581] enterPrevote: ProposalBlock is invalid       module=consensus height=12 round=15 err="wrong Block.Header.AppHash.  Expected 2D9156F449BB25D1BC87776D7FDAF644ABFCC51898B778A4167D31973F1274C9, got 261B01FFD130716F4E21C7A891F3400EA39F117D1A01B029E6A74483038E8CB5"
```

**Comment by andrewnguyen22** [comment]
Potential cause is the use of the DB.Iterator in GetMatureClaims in SendProofTx()

**Comment by andrewnguyen22** [comment]
Closed with 232efa370c20ebfd90ce41fd7379f5ca119a3b57
